templates/commands/render_action_regexnamelookup_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"testing"
+
+	"github.com/abcxyz/abc/templates/model"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestActionRegexNameLookup(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name         string
+		inputs       map[string]string
+		initContents map[string]string
+		rr           *model.RegexNameLookup
+		want         map[string]string
+		wantErr      string
+	}{
+		{
+			name: "simple_success",
+			initContents: map[string]string{
+				"a.txt": "alpha beta gamma delta",
+			},
+			inputs: map[string]string{
+				"my_input": "foo",
+			},
+			rr: &model.RegexNameLookup{
+				Paths: modelStrings([]string{"."}),
+				Replacements: []*model.RegexNameLookupEntry{
+					{
+						Regex: model.String{Val: `\b(?P<my_input>b...) (?P<my_input>g....)`},
+					},
+				},
+			},
+			want: map[string]string{
+				"a.txt": "alpha foo foo delta", //nolint:dupword
+			},
+		},
+		{
+			name: "missing_template_variable_should_fail",
+			initContents: map[string]string{
+				"a.txt": "alpha beta gamma",
+			},
+			inputs: map[string]string{},
+			rr: &model.RegexNameLookup{
+				Paths: modelStrings([]string{"."}),
+				Replacements: []*model.RegexNameLookupEntry{
+					{
+						Regex: model.String{Val: "(?P<mysubgroup>beta)"},
+					},
+				},
+			},
+			want: map[string]string{
+				"a.txt": "alpha beta gamma",
+			},
+			wantErr: `there was no template input variable matching the subgroup name "mysubgroup"`,
+		},
+		{
+			name: "named_group_autolookup_should_reject_unnamed_subgroups",
+			initContents: map[string]string{
+				"a.txt": "alpha beta gamma",
+			},
+			inputs: map[string]string{
+				"my_input": "foo",
+			},
+			rr: &model.RegexNameLookup{
+				Paths: modelStrings([]string{"."}),
+				Replacements: []*model.RegexNameLookupEntry{
+					{
+						Regex: model.String{Val: `\b(?P<my_input>b...) (g....)`},
+					},
+				},
+			},
+			want: map[string]string{
+				"a.txt": "alpha beta gamma",
+			},
+			wantErr: `must be named`,
+		},
+		{
+			name: "template_expr_in_regex_and_groupname_should_work",
```

Comment:
Super nit: I'm not sure "should_work" is helpful in the name.

Reply:
Let me send a follow PR to fix this, I've been doing this throughout this codebase.